packages/payment-detection/src/erc777/superfluid-detector.ts

```diff
@@ -21,15 +22,65 @@ export class SuperFluidPaymentDetector extends ReferenceBasedDetector<
     super(PaymentTypes.PAYMENT_NETWORK_ID.ERC777_STREAM, advancedLogic.extensions.erc777Stream);
   }
 
+  protected isSubsequentRequest(request: RequestLogicTypes.IRequest): boolean {
+    return !!request.extensions[ExtensionTypes.ID.PAYMENT_NETWORK_ERC777_STREAM].values
+      .masterRequestId;
+  }
+
   /**
-   * Extracts the balance and events of an address
+   * This returns the specific values we store for the ERC777 extensions.
+   * The extra fields specific for this extension are:
+   * expectedStartDate: The expected start date of streaming
+   * expectedFlowRate: extensionAction.parameters.expectedFlowRate,
+   * previousRequestId: The previous requestId in the streaming series
+   * masterRequestId: The first requestId of the streaming series
+   * recurrenceNumber: The position of this request in the streaming series, 0 indexed.
+   * @param request The request we are processing
+   * @returns Object
+   */
+  protected getSubsequentValues(request: RequestLogicTypes.IRequest): Record<string, any> {
+    return request.extensions[ExtensionTypes.ID.PAYMENT_NETWORK_ERC777_STREAM].values;
+  }
+
+  public async getBalance(
+    request: RequestLogicTypes.IRequest,
+  ): Promise<
+    PaymentTypes.IBalanceWithEvents<
+      PaymentTypes.IERC777PaymentEventParameters | PaymentTypes.IDeclarativePaymentEventParameters
+    >
+  > {
+    const totalBalance = await super.getBalance(request);
+    if (totalBalance.error) {
+      return totalBalance;
+    }
+    if (totalBalance.balance) {
+      let expectedPreviousBalance = 0;
+      if (this.isSubsequentRequest(request)) {
+        const subrequestValues = this.getSubsequentValues(request);
+        expectedPreviousBalance =
+          subrequestValues.recurrenceNumber * parseFloat(request.expectedAmount.toString());
+      }
+      const remainingBalance = parseFloat(totalBalance.balance) - expectedPreviousBalance;
+      let requestBalance = 0;
+      if (remainingBalance > request.expectedAmount) {
+        requestBalance = parseFloat(request.expectedAmount.toString());
+      } else {
+        requestBalance = remainingBalance > 0 ? remainingBalance : 0;
```

Comment:
Shouldn't you make all those computations with `BigNumber`?

Reply:
Done. I keep forgetting about BigNumber